New ChestMenu option

VoidAngel opened the pull request:
Mostly just to remove the annoying metrics spam D:

VoidAngel committed:
Switch to bStats

Mostly just to remove the annoying metrics spam D:

TheBusyBiscuit commented:
What spam?

VoidAngel commented:
Ah sorry just realized the source of the spam was a 3rd party plugin pointing at CS-CoreLib. 
I initially thought it was just because mcstats is dead.

TheBusyBiscuit commented:
Alright, but as far as I know it isn't dead, just unavailable for most of the time.
However I also heard that the domain expires this month, so a switch may be an option.

I will look into bstats.

TheBusyBiscuit commented on src/org/bstats/Metrics.java:
```diff
@@ -0,0 +1,667 @@
+        if (System.getProperty("bstats.relocatecheck") == null || !System.getProperty("bstats.relocatecheck").equals("false")) {
+            // Maven's Relocate is clever and changes strings, too. So we have to use this little "trick" ... :D
+            final String defaultPackage = new String(
+                    new byte[]{'o', 'r', 'g', '.', 'b', 's', 't', 'a', 't', 's', '.', 'b', 'u', 'k', 'k', 'i', 't'});
```
It throws an exception when not placed in "org.bstats.bukkit"
It isn't placed there.

VoidAngel replied:
Are you sure? I was under the impression that it only threw the exception if the package string was equal to "org.bstats.bukkit" or "your.package"
`            if (Metrics.class.getPackage().getName().equals(defaultPackage) || Metrics.class.getPackage().getName().equals(examplePackage)) {
                throw new IllegalStateException("bStats Metrics class has not been relocated correctly!");
            }`

TheBusyBiscuit replied:
Aaaaaand that is why I shouldn't be at my pc at 10 am.

VoidAngel commented:
Anywayssss no pressure xD Now that I see this isn't the source of my metrics spam, I know this doesn't really affect server owners, so I'll leave it up to you

VoidAngel committed:
Adds a new characteristic to ChestMenus

This adds the ability to specify whether empty slots in a given ChestMenu are clickable. This PR will be paired with another in the Slimefun4 repository.

VoidAngel commented:
PS I can remove the bstats thing if you want.

TheBusyBiscuit commented:
Your ChestMenu implementation looks good, exactly what I meant.

I haven't made up my mind on bstats yet but I would prefer to seperate these two by moving bstats on a different branch and making a Pull Request that can sit around till I made a decision.

VoidAngel committed:
Revert "Switch to bStats"

This reverts commit 54b358a82902eb26d2054f645dd1b9ffa51d1e19.
